static/js/components/forms/WebsiteCollectionItemForm.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useCallback, useState } from "react"
+import { Form, Formik, Field, FormikHelpers } from "formik"
+
+import SelectField, { Option } from "../widgets/SelectField"
+import { WCItemCreateFormFields } from "../../types/forms"
+import { Website } from "../../types/websites"
+import { Entities } from "redux-query"
+import { ActionPromise } from "redux-query-react"
+import { siteApiListingUrl } from "../../lib/urls"
+import { WebsiteListingResponse } from "../../query-configs/websites"
+
+interface Props {
+  onSubmit: (value: WCItemCreateFormFields) => ActionPromise<Entities>
+}
+
+const formatOptions = (websites: Website[]) =>
+  websites.map(website => ({ label: website.title, value: website.uuid }))
+
+export default function WebsiteCollectionItemForm(props: Props): JSX.Element {
+  const { onSubmit } = props
+  const [options, setOptions] = useState<Option[]>([])
+
+  const onSubmitCB = useCallback(
+    (
+      values: WCItemCreateFormFields,
+      formikHelpers: FormikHelpers<WCItemCreateFormFields>
+    ) => {
+      const { resetForm } = formikHelpers
+      onSubmit(values)
+      resetForm()
+    },
+    [onSubmit]
+  )
+
+  const loadOptions = useCallback(
+    async (inputValue: string, callback: (options: Option[]) => void) => {
+      const url = siteApiListingUrl
+        .query({ offset: 0 })
+        .param({ search: inputValue })
+        .toString()
+
+      const response = await fetch(url)
+      const json: WebsiteListingResponse = await response.json()
+      const { results } = json
+      const options = formatOptions(results)
+      setOptions(current => [...current, ...options])
+      callback(options)
+    },
+    [setOptions]
+  )
+
+  return (
+    // @ts-ignore
+    <Formik onSubmit={onSubmitCB} initialValues={{ website: undefined }}>
```

Comment:
What does the `website: undefined` do here?

Reply:
sort of a weird thing, I'm going to probably add a comment to clarify

basically Formik and react-select disagree here! Formik thinks that `initialValues` should have `""`, but react-select does not consider `""` to be an empty value (afaict). So before I was passing `{website: "" }` as the `initialValues` prop, but this was causing the placeholder not to show up in react-select. Once I changed it to `undefined` the react-select placeholder started to show up. The `@ts-ignore` is here because Formik wants this prop to be `Record<string, string>` I think (or something along those lines).